test/definitions/amount_energy_from_mass_and_temperature_test.py

```diff
@@ -0,0 +1,66 @@
+from collections import namedtuple
+from pytest import approx, fixture, raises
+
+from symplyphysics import (
+    units, convert_to, SI, errors
+)
+from symplyphysics.definitions import amount_energy_from_mass_and_temperature as amount_energy
+
+# How much energy does it take to heat 0.5 liter of water to 50 Celsius degree?
+# Specific heat capacity of water is 4200 J/kg*K , ignore losses.
+# Density of water - 1000 kilogram / meter**3, 273 Kelvin == 0 Celsius_degree
+CELSIUS_TO_KELVIN_OFFSET = int(273)
+@fixture
+def test_args():
+    C = units.Quantity('C')
+    SI.set_quantity_dimension(C, units.energy / (units.mass * units.temperature))
+    SI.set_quantity_scale_factor(C, 4200 * units.joule / (units.kilogram * units.kelvin))
+    m = units.Quantity('m')
+    SI.set_quantity_dimension(m, units.mass)
+    SI.set_quantity_scale_factor(m, 0.5 * units.liter * 1000 * units.kilogram / units.meter**3)
+    t1 = units.Quantity('t1')
+    SI.set_quantity_dimension(t1, units.temperature)
+    SI.set_quantity_scale_factor(t1, (273 - CELSIUS_TO_KELVIN_OFFSET) * units.kelvin)
```

Comment:
Are you heating from 0 kelvin temperature? I suggest to drop celsius degrees here and use kelvin. Tests are kinda synthetic so you can precalc some values for shorter code.